dotcom-rendering/src/web/components/SignInGate/types.ts

```diff
@@ -18,14 +19,44 @@ export type SignInGateComponent = {
 	}: CanShowGateProps) => Promise<boolean>;
 };
 
-export interface SignInGateProps {
+export const ALL_USER_TYPES = ['new', 'guest', 'current'] as const;
+type UserTypeTuple = typeof ALL_USER_TYPES;
+export type UserType = UserTypeTuple[number];
+
+export const isUserType = (value: unknown): value is UserType =>
+	ALL_USER_TYPES.includes(value as UserType);
+
+export const ALL_PRODUCTS = [
+	'Contribution',
+	'DigitalPack',
+	'Paper',
+	'GuardianWeekly',
+] as const;
+type ProductTuple = typeof ALL_PRODUCTS;
+export type Product = ProductTuple[number];
+
+export const isProduct = (value: unknown): value is Product =>
+	ALL_PRODUCTS.includes(value as Product);
+export interface CheckoutCompleteCookieData {
+	userType: UserType;
+	product: Product;
+}
+
+export function isCheckoutCompleteCookieData(
+	obj: unknown,
+): obj is CheckoutCompleteCookieData {
+	return isObject(obj) && isUserType(obj.userType) && isProduct(obj.product);
+}
+export type SignInGateProps = {
 	signInUrl: string;
 	guUrl: string;
 	dismissGate: () => void;
 	ophanComponentId: string;
 	abTest?: CurrentSignInGateABTest;
 	isMandatory?: boolean;
-}
+	checkoutCompleteCookieData?: CheckoutCompleteCookieData;
+	personaliseSignInGateAfterCheckoutSwitch?: boolean;
+};
```

Comment:
@OllysCoding simplified the type system by just making these optional fields on the same interface. 

It means doing an additional type check here: https://github.com/guardian/dotcom-rendering/pull/7222/files#diff-04d09f3aeb1e9344cc0053d2052dacdc0f01fbb956830a4b16103881d28b4555R181 
but I think this is ok, and makes the typing a lot simpler